sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -418,17 +405,37 @@ export class EventHubReceiver extends LinkEntity {
       await this._context.readyToOpenLink({ abortSignal });
       await this._negotiateClaim({ setTokenRenewal: false, abortSignal, timeoutInMs });
 
-      const receiverOptions: CreateReceiverOptions = {
+      const options: RheaReceiverOptions = {
+        name: this.name,
+        autoaccept: true,
+        source: {
+          address: this.address,
+        },
+        credit_window: 0,
         onClose: (context) => this._onAmqpClose(context),
         onError: (context) => this._onAmqpError(context),
         onMessage: (context) => this._onAmqpMessage(context),
         onSessionClose: (context) => this._onAmqpSessionClose(context),
         onSessionError: (context) => this._onAmqpSessionError(context),
       };
-      if (this.checkpoint > -1) {
-        receiverOptions.eventPosition = { sequenceNumber: this.checkpoint };
+
+      if (typeof this.ownerLevel === "number") {
+        options.properties = {
+          [Constants.attachEpoch]: types.wrap_long(this.ownerLevel),
+        };
+      }
+
+      if (this.options.trackLastEnqueuedEventProperties) {
+        options.desired_capabilities = Constants.enableReceiverRuntimeMetricName;
       }
-      const options = this._createReceiverOptions(receiverOptions);
+
+      const eventPosition =
+        this.checkpoint > -1 ? { sequenceNumber: this.checkpoint } : this.eventPosition;
+      // Set filter on the receiver if event position is specified.
+      const filterClause = getEventPositionFilter(eventPosition);
+      (options.source as Source).filter = {
+        "apache.org:selector-filter:string": types.wrap_described(filterClause, 0x468c00000004),
+      };
```

Comment:
`eventPosition` and `filterClause` are guaranteed to be defined so no need to check whether they're defined first. Also `options.source` should be typed as Source not as `any`.